Fix misplaced deprecation attribute on vtkCellArray::VisitState

The VTK_DEPRECATED_IN_9_8_0(...) macro expands to a C++ [[deprecated]]
attribute. Placing it before the `template` keyword of a class template is
ill-formed; the attribute must follow the class-key.

--- Common/DataModel/vtkCellArray.h
+++ Common/DataModel/vtkCellArray.h
@@ -999,9 +999,8 @@
   ///@}
 
   // Holds connectivity and offset arrays of the given ArrayType.
-  VTK_DEPRECATED_IN_9_8_0("Use DispatchUtilities")
   template <typename ArrayT>
-  struct VisitState
+  struct VTK_DEPRECATED_IN_9_8_0("Use DispatchUtilities") VisitState
   {
     using ArrayType = ArrayT;
     using ValueType = typename ArrayType::ValueType;
